### PR TITLE
ENH: Support gumbel_r in scipy.stats.anderson

### LINKED ISSUE #2592: scipy.stats.anderson assumes gumbel_l
From inspecting the code, gumbel_l is assumed, without any documentation. There should be two flags, gumbel_l and gumbel_r, accepted by scipy.stats.anderson. Then, the anderson test function should use the correct gumbel distribution. The documentation could benefit from a couple of examples and some clarification on this point, too.

I would make a pull request myself, but I don't have time right now.

### PR DESCRIPTION
**Pull request by saketkc**
Closes #2592

### CODE REVIEW

**Comment by saketkc** [comment]
Good to merge?

File: scipy/stats/morestats.py [hunk 1]
```diff
@@ -1082,9 +1082,9 @@ def anderson(x, dist='norm'):
            pp. 591-595.
 
     """
-    if dist not in ['norm', 'expon', 'gumbel', 'extreme1', 'logistic']:
+    if dist not in ['norm', 'expon', 'gumbel_l', 'gumbel_r', 'extreme1', 'logistic']:
```
Comment (rgommers) [inline scipy/stats/morestats.py]: looks like ``gumbel`` should be kept here as an alias of ``gumbel_l``, to not break backwards compatibility? 

It doesn't need to be in the docstring.

File: scipy/stats/tests/test_morestats.py [hunk 2]
```diff
@@ -79,6 +79,14 @@ def test_expon(self):
             np.seterr(**olderr)
         assert_(A > crit[-1])
 
+    def test_gumbel(self):
+        rs = RandomState(1234567890)
+        x1 = rs.gumbel(size=50)
+        A,crit,sig = stats.anderson(x1, 'gumbel_l')
```
Comment (rgommers) [inline scipy/stats/tests/test_morestats.py]: pep8: spaces after commas. same two lines lower

**Comment by rgommers** [comment]
Not necessary to merge but good for bonus point: adding an example to the docstring.

File: scipy/stats/tests/test_morestats.py [hunk 3]
```diff
@@ -79,6 +79,14 @@ def test_expon(self):
             np.seterr(**olderr)
         assert_(A > crit[-1])
 
+    def test_gumbel(self):
+        rs = RandomState(1234567890)
```
Comment (josef-pkt) [inline scipy/stats/tests/test_morestats.py]: I think more than 6 digits for the random seed was causing an error on some machines (32bit?)

**Comment by josef-pkt** [comment]
(I don't know whether this works correctly this way based on jjust reading the code.)

**Comment by saketkc** [comment]
I am adding Examples. The test borrowed code and seed from the previous tests for normal, exponential.

**Comment by saketkc** [comment]
Will rebase, once it is good enough to merge.

**Comment by saketkc** [comment]
@rgommers @josef-pkt Would you like to go review this once again?

**Comment by saketkc** [comment]
@josef-pkt @rgommers Would you be able to review this?

**Comment by josef-pkt** [comment]
Don't do whitespace cleanup in the same commit (or even better not even in the same PR) as substantial changes.
It's impossible to see what has changed in the test file.

**Comment by saketkc** [comment]
The only real change in the test file is this function:
https://github.com/scipy/scipy/pull/4508/files#diff-d723d053b57ebaeb021ea79bb4f0df1bR84
The rest of it is cosmetic, PEP8 stuff.

File: scipy/stats/tests/test_morestats.py [hunk 4]
```diff
@@ -57,33 +58,43 @@ def test_bad_arg(self):
...
             np.seterr(**olderr)
         assert_(A > crit[-1])
 
+    def test_gumbel(self):
+        rs = RandomState(12345)
+        x1 = rs.gumbel(size=50)
+        A, crit, sig = stats.anderson(x1, 'gumbel_l')
+        assert_array_less(crit[:-1], A)
+        A, crit, sig = stats.anderson(x1, 'gumbel_r')
+        assert_array_less(A, crit[-2:])
```
Comment (josef-pkt) [inline scipy/stats/tests/test_morestats.py]: I think the test should follow the pattern of the other tests and sample from the Null distribution, i.e. `gumbel_r`.

AFAIR, to check that these functions are correct, I ran a small Monte Carlo. Generate 1000 samples of gumbel_r and check that the rejection rate is approximately the right one.
Those checks/tests are not in the scipy test suite.

**Comment by josef-pkt** [comment]
The change to anderson looks good to me. However, since we don't have verified comparison numbers, I think we should run a Monte Carlo to check that the gumbel_r really has the same critical values. I assume it does.

**Comment by saketkc** [comment]
I will do that, especially in the wake of https://github.com/scipy/scipy/issues/1624

**Comment by josef-pkt** [comment]
good you found 1624, I didn't remember anything.

**Comment by rgommers** [comment]
@saketkc did you run that check? 

I also make some style changes here: https://github.com/rgommers/scipy/tree/pr/4508 - maybe you can take over that branch (it's rebased).

**Comment by saketkc** [comment]
On it @rgommers

**Comment by saketkc** [comment]
I haven't been able to make any progress on this, so I am going to close it for now. I will probably open a new PR for this, soon.

**Pull request closed without merging**